src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint-disable react/button-has-type */
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  todos: Todo[],
+  onToggleTodo: (id: number) => void;
+  onDeleteTodo: (id: number) => void;
+  setAllTodos: (todos: Todo[]) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  todos,
+  onToggleTodo,
+  onDeleteTodo,
+  setAllTodos,
+}) => {
+  const { id, title, completed } = todo;
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(title);
+
+  const handleToggleTodo = (idT: number) => {
+    onToggleTodo(idT);
+  };
+
+  const handleDeleteTodo = (idT: number) => {
```

Comment:
What is `idT`? It is better to give more meaningful names for props.
Also, you can use functions from props directly